Lib/argparse.py

```diff
@@ -1360,7 +1360,7 @@ def __init__(self,
         self._defaults = {}
 
         # determines whether an "option" looks like a negative number
-        self._negative_number_matcher = _re.compile(r'^-\d[\d_]*(\.\d[\d_]*)?$')
+        self._negative_number_matcher = _re.compile(r'^-(\d+(_\d+)*(\.\d+(_\d+)*)?|\.\d+(_\d+)*)$')
```

Comment:
The pattern looks fine for capturing negatives without a leading digit, but it does have a (rather silly) breaking change: `-12__3` was accepted by the old pattern but not the new.

I think support for runs of underscores was accidental, though (and undocumented), so it seems fine to remove it. Would be remiss not to point it out, though!

A